modules: hal_nxp: osa: Remove unused OSA API

Radimli opened the pull request:
- Deleted file fsl_os_abstraction.h is unused and deprecated
- Location of current OSA API available in modules/hal/nxp/mcux/mcux-sdk/components/osa

danieldegrasse reviewed (commented):
Thanks for this change- please take a look at your `git` configuration. There are a few issues the compliance script is flagging (you can view these by clicking "Details" on the failed compliance check)

- Commit author does not match the `signed-off` entry: You can update the author of this commit with `git commit --amend --author "Radim.Lipka@nxp.com"`
- Commit body is too long (>75 characters on one line)- just add a line break in the commit body to fix this, editors like vim will do this for you automatically

Edit: we will also need to look at the failing twister tests- these point to places where the OSA API might still be used. It looks like this PR has some USB sample build failures.

Radimli commented:
> Thanks for this change- please take a look at your `git` configuration. There are a few issues the compliance script is flagging (you can view these by clicking "Details" on the failed compliance check)
> 
> * Commit author does not match the `signed-off` entry: You can update the author of this commit with `git commit --amend --author "Radim.Lipka@nxp.com"`
> * Commit body is too long (>75 characters on one line)- just add a line break in the commit body to fix this, editors like vim will do this for you automatically
> 
> Edit: we will also need to look at the failing twister tests- these point to places where the OSA API might still be used. It looks like this PR has some USB sample build failures.

Hello @danieldegrasse,

yes, cmake in USB MW should be modified. Already discussing it with @MarkWangChinese

Radimli commented:
@danieldegrasse @MarkWangChinese 
It should be solved in 
https://github.com/zephyrproject-rtos/hal_nxp/pull/433

MarkWangChinese commented:
Hi @Radimli You can update this PR (changing the west.yml) to contain the https://github.com/zephyrproject-rtos/hal_nxp/pull/433 like: https://github.com/zephyrproject-rtos/zephyr/pull/77973/files#diff-d7f4082837dc3cb270fcf1abd68188216fdadee294e01611b8ca59ddea7fdbb3R201, then let the CI to check it.

danieldegrasse commented on west.yml:
```diff
@@ -198,7 +198,7 @@ manifest:
         - hal
     - name: hal_nxp
-      revision: 466000a80e6eb5bbcb691bae936ec5654a7796d4
+      revision: 7a17a78b14565a12a6dc5e527f71c351ed4df6b6
```
```suggestion
      revision: pull/433/head
```
Don't grab the new SHA directly, that won't be possible until the PR in the HAL merges. Instead, use syntax like I suggested above so west can pull in the contents of the PR

danieldegrasse commented:
Build tests look good, but you still have a compliance error. If you want to check this locally rather than waiting on CI (and are using Linux) you can run `./scripts/ci/check_compliance.py -c $(git merge-base HEAD main)...HEAD` from your zephyr repo, with this branch checked out.

Radimli committed:
modules: hal_nxp: osa: Remove deprecated OSA API

New location of current OSA API available in
/hal/nxp/mcux/mcux-sdk/components/osa

Signed-off-by: Radim Lipka <Radim.Lipka@nxp.com>

danieldegrasse commented:
Please address merge conflict when you get a chance

danieldegrasse commented:
@Radimli ping to fix merge conflict

dleach02 commented:
@Radimli we are working to sequence the HAL PRs
